src/main/java/net/minecraftforge/server/command/CommandTreeBase.java

```diff
@@ -51,7 +52,19 @@ public void addSubcommand(ICommand c)
 
     public Collection<ICommand> getSubCommands()
     {
-        return Collections.unmodifiableCollection(commandMap.values());
+        return getCommandMap().values();
+    }
+
+    public Map<String, ICommand> getCommandMap()
+    {
+        return Collections.unmodifiableMap(commandMap);
```

Comment:
Its not that important, but for future reference if you just want a unmodifable view, you can call this function once and cache the return. If the underlying map is update the view will be updated. Such is the power of views instead of deep copies.